panoramisk/manager.py

```diff
@@ -200,8 +202,24 @@ def connect(self):
                 ssl=self.config['ssl']),
             loop=self.loop)
         t.add_done_callback(self.connection_made)
+
+        if run_forever:
+            self.run_forever(on_startup, on_shutdown)
         return t
 
+    def run_forever(self, on_startup, on_shutdown):
+        """Start loop forever"""
+        try:
+            if on_startup:
+                self.loop.run_until_complete(on_startup(self))
+            self.loop.run_forever()
+        except (KeyboardInterrupt, SystemExit):
+            self.close()
+        finally:
+            if on_shutdown:
+                self.loop.run_until_complete(on_shutdown(self))
```

Comment:
same here. you should create a task and use `add_done_callback(lambda *args: self.loop.close())`